django/contrib/admin/options.py

```diff
@@ -2138,17 +2138,20 @@ def has_delete_permission(self, request, obj=None):
         return super().has_delete_permission(request, obj)
 
     def has_view_permission(self, request, obj=None):
-        if self.opts.auto_created:
-            opts = self.opts
+        opts = self.opts
+        if opts.auto_created:
             # The model was auto-created as intermediary for a many-to-many
             # Many-relationship; find the target model.
             for field in opts.fields:
                 if field.remote_field and field.remote_field.model != self.parent_model:
                     opts = field.remote_field.model._meta
                     break
+
+            codename_view = get_permission_codename('view', opts)
+            codename_change = get_permission_codename('change', opts)
             return (
-                request.user.has_perm('%s.%s' % (opts.app_label, get_permission_codename('view', opts))) or
-                request.user.has_perm('%s.%s' % (opts.app_label, get_permission_codename('change', opts)))
+                request.user.has_perm('%s.%s' % (opts.app_label, codename_view)) or
+                request.user.has_perm('%s.%s' % (opts.app_label, codename_change))
```

Comment:
Both these lines come in at under 120 chars, so there's no real need for the temporary variables.